src/object/srv_obj_migrate.c

```diff
@@ -1401,8 +1400,22 @@ migrate_fetch_update_bulk(struct migrate_one *mrone, daos_handle_t oh,
 			rc = __migrate_fetch_update_bulk(mrone, oh, &iod, 1,
 							 mrone->mo_iods_update_ephs[i][j],
 							 DIOF_FOR_MIGRATION, ds_cont);
-			if (rc > 0)
-				D_GOTO(out, rc);
+			if (rc < 0) {
+				if (rc == -DER_VOS_PARTIAL_UPDATE) {
+					/* In some cases, EC aggregation failed to delete the
+					 * replicate recx, then during rebuild these replicate
+					 * recx might be rebuilt again. Since the data rebuilt
+					 * from parity will be rebuilt first. so let's ingore
+					 * this replicate recx for now.
+					 */
+					D_WARN(DF_UOID" "DF_RECX"/"DF_X64" already rebuilt\n",
+					       DP_UOID(mrone->mo_oid), DP_RECX(iod.iod_recxs[0]),
+					       mrone->mo_iods_update_ephs[i][j]);
+					rc = 0;
```

Comment:
since now rebuild and ec agg cannot happen concurrently, do we still need this handling?

Reply:
hmm, if replica extent is left there for some reasons, then it might still see this issue, so we still need handling in this case.

Reply:
@wangdi1 sorry you mean for the case that  for example during EC agg, it will 1) generate new parity extend, and 2) remove old replica extend. But somehow the step 2) is not done and rebuild cut in and here try to rebuild the replica extend? the build's local vos update epoch == the exist replica extends (left as step 2 did not remove it) and extend range overlapped with each other?

will the EC agg exit executing (after finish step 2) and then rebuild should not hit that case?